org.eclipse.paho.android.service/src/main/java/org/eclipse/paho/android/service/MqttTokenAndroid.java

```diff
@@ -126,6 +128,17 @@ void notifyComplete() {
     }
   }
 
+  /**
+   * notify successful completion of the subscribe operation
+   */
+  void notifySubscription(List<String> successTopics, List<String> failedTopics) {
+    synchronized (waitObject) {
```

Comment:
@rubenquadros12 what is `waitObject` here?

Reply:
It is used to lock the thread during operation. This is an existing codebase so have replicated the method.